elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -417,7 +424,14 @@ public void performOperations(List<LuceneWork> workList, IndexingMonitor monitor
 			elasticsearchWorks.add( luceneWork.acceptIndexWorkVisitor( visitor, monitor ) );
 		}
 
-		workProcessor.executeSyncSafe( elasticsearchWorks );
+		if ( sync ) {
+			workProcessor.executeSyncSafe( elasticsearchWorks );
+		}
+		else {
+			for ( ElasticsearchWork<?> work : elasticsearchWorks ) {
```

Comment:
I wouldn't consider this a blocker and not inclined to invest too much time on this as we want to evolve the ErrorHandler API.

Since errors are handled individually, users aren't going to miss them.

Just add a comment?